quizzes/views.py

```diff
@@ -0,0 +1,223 @@
+from django.http import HttpResponseRedirect
+from django.shortcuts import render, get_object_or_404, redirect
+from django.urls import reverse
+from django.views import generic
+from django.utils import timezone
+from .models import Quiz, Category, Question, Answer, Feedback
+from django.contrib.sessions.backends.db import SessionStore
+
+
+# Built following alongside Django Software Foundation's Writing your first Django app Tutorial
+# https://docs.djangoproject.com/en/3.0/intro/tutorial01/
+
+class IndexView(generic.ListView):
+    template_name = 'quizzes/index.html'
+    context_object_name = 'latest_quiz_list'
+
+    def get_queryset(self):
+        """Return the last five published quizzes"""
+        return Quiz.objects.filter(pub_date__lte=timezone.now()).order_by('-pub_date')[:5]
+
+
+class QuizDetailView(generic.DetailView):
+    model = Quiz
+    template_name = 'quizzes/quiz_detail.html'
+
+    def get_queryset(self):
+        """Return the last five published quizzes"""
+        return Quiz.objects.filter(pub_date__lte=timezone.now())
+
+
+class CategoryDetailView(generic.DetailView):
+    model = Category
+    template_name = 'quizzes/category_detail.html'
+
+
+class QuestionDetailView(generic.DetailView):
+    model = Question
+    template_name = 'quizzes/question_detail.html'
+
+
+def feedback(request, quiz_id):
+    quiz = get_object_or_404(Quiz, pk=quiz_id)
+    return render(request, 'quizzes/feedback.html', {
+        'quiz': quiz,
+        'feed_dict': request.session[quiz.session_feedback()],
+        'norm_scores': request.session[quiz.session_norm_data()],
+        })
+
+
+def new_quiz(request, quiz_id, category_id):
+    """
+    Function to create a new session when "Take Quiz" is selected.
+    """
+    quiz = get_object_or_404(Quiz, pk=quiz_id)
+    quiz_id = quiz.id
+    categories = quiz.get_quiz_categories()
+    category_id = quiz.category_set.first().id
+    questions = quiz.get_quiz_questions()
+    question_id = quiz.question_set.first().id
+    question_list = [q.id for q in questions]
+
+    # Reset Quiz session
```

Comment:
It might be worth creating subfunctions for some of the these independent actions (resetting session, resetting category score, etc.)

Reply:
Ah, there are some artifacts in here of the old functionality that I forgot to delete (resetting the category score and answer_selected). Will update for next PR.